tests/plugins/clipboard/dropImage.js

```diff
@@ -0,0 +1,169 @@
+
+			setHtmlWithSelection( this.editor, '<p class="p">Paste image here:^</p>' );
+			this.assertDropImage( dropEvt, imageType, expected, {
+				dropContainer: this.editor.editable().findOne( '.p' ).getChild( 0 ),
+				dropOffset: 17
+			} );
+		},
+
+		'test drop .jpeg image': function() {
+			var dropEvt = bender.tools.mockDropEvent(),
+				imageType = 'image/jpeg',
+				expected = '<p class="p">Paste image here:<img src="data:image/jpeg;base64,fileMockBase64=" /></p>';
+
+			FileReader.setFileMockType( imageType );
+			FileReader.setReadResult( 'load' );
+
+			setHtmlWithSelection( this.editor, '<p class="p">Paste image here:^</p>' );
+			this.assertDropImage( dropEvt, imageType, expected, {
+				dropContainer: this.editor.editable().findOne( '.p' ).getChild( 0 ),
+				dropOffset: 17
+			} );
+		},
+
+		'test drop .gif image': function() {
+			var dropEvt = bender.tools.mockDropEvent(),
+				imageType = 'image/gif',
+				expected = '<p class="p">Paste image here:<img src="data:image/gif;base64,fileMockBase64=" /></p>';
+
+			FileReader.setFileMockType( imageType );
+			FileReader.setReadResult( 'load' );
+
+			setHtmlWithSelection( this.editor, '<p class="p">Paste image here:^</p>' );
+			this.assertDropImage( dropEvt, imageType, expected, {
+				dropContainer: this.editor.editable().findOne( '.p' ).getChild( 0 ),
+				dropOffset: 17
+			} );
+		},
+
+		'test drop unsupported image type': function() {
+			var dropEvt = bender.tools.mockDropEvent(),
+				imageType = 'application/pdf',
+				expected = '<p class="p">Paste image here:</p>';
+
+			FileReader.setFileMockType( imageType );
+			FileReader.setReadResult( 'load' );
+
+			setHtmlWithSelection( this.editor, '<p class="p">Paste image here:^</p>' );
+			this.assertDropImage( dropEvt, imageType, expected, {
+				dropContainer: this.editor.editable().findOne( '.p' ).getChild( 0 ),
+				dropOffset: 17
+			} );
+		},
+
+		'test abort drop': function() {
+			var dropEvt = bender.tools.mockDropEvent(),
+				imageType = 'image/png',
+				expected = '<p class="p">Paste image here:</p>';
+
+			FileReader.setFileMockType( imageType );
+			FileReader.setReadResult( 'abort' );
+
+			setHtmlWithSelection( this.editor, '<p class="p">Paste image here:^</p>' );
+			this.assertDropImage( dropEvt, imageType, expected, {
+				dropContainer: this.editor.editable().findOne( '.p' ).getChild( 0 ),
+				dropOffset: 17
+			} );
+		},
+
+		'test failed drop': function() {
+			var dropEvt = bender.tools.mockDropEvent(),
+				imageType = 'image/gif',
+				expected = '<p class="p">Paste image here:</p>';
+
+			FileReader.setFileMockType( imageType );
+			FileReader.setReadResult( 'error' );
+
+			setHtmlWithSelection( this.editor, '<p class="p">Paste image here:^</p>' );
+			this.assertDropImage( dropEvt, imageType, expected, {
+				dropContainer: this.editor.editable().findOne( '.p' ).getChild( 0 ),
+				dropOffset: 17
+			} );
+		},
+
+		assertDropImage: function( evt, type, expected, config ) {
```

Comment:
We do not keep such functions inside tests object (however such a practice can be seen in some of old tests). Please move it outside the object, as a normal function.